tests/func/test_dvcfile.py

```diff
@@ -285,6 +285,32 @@ def test_remove_stage_preserves_comment(tmp_dir, dvc, run_copy):
     )
 
 
+def test_remove_stage_removes_dvcfiles_if_no_stages_left(tmp_dir, dvc):
+    tmp_dir.gen(
+        "dvc.yaml",
+        textwrap.dedent(
+            """\
+            stages:
+                generate-foo:
+                    cmd: "echo foo > foo"
+                    outs:
+                    - foo"""
+        ),
```

Comment:
@nik123, we don't care about yaml structure here. It'd be better either to create stage via `run_copy` fixture or use `dump_yaml`.

Reply:
Fixed